packages/fuels-contract/src/script.rs

```diff
@@ -403,6 +406,91 @@ impl Script {
     }
 }
 
+/// Based on the returned Contract's output_params and the receipts returned from a call,
+/// decode the values and return them.
+pub fn get_decoded_output(
+    output_param: ParamType,
+    receipts: &mut Vec<Receipt>,
+) -> Result<Token, Error> {
+    // Multiple returns are handled as one `Tuple` (which has its own `ParamType`)
+    let (encoded_value, index) = match output_param.get_return_location() {
+        ReturnLocation::ReturnData => {
+            match receipts.iter().find(|&receipt| {
+                matches!(receipt,
+                    Receipt::ReturnData { id: _, data, .. } if !data.is_empty())
```

Comment:
This seems to have been copied before a recent bug fix that includes a check for the contract_id here.

Reply:
There is no contract id here as we're executing a script, unless I misunderstood how the contract ids work

Reply:
Keep in mind for the refactor, this can potentially lead to the wrong value being decoded when we call multiple contracts via the script.